### PR TITLE
ChemNet Fixes and Additions

### PR DESCRIPTION
**Pull request by VIGS25**
This PR adds in enhancements in line with the ChemNet paper:

**Transformers**:
1. MinMax transformer is added along with tests and documentation - This is the Transformer used in ChemNet pretraining. 

**Tox21, HIV, ChEMBL25, FreeSolv**
1. Train, Valid and test split fractions added in kwargs for all the four datasets
2. RandomStratifiedSplitter added in Tox21 and HIV, since that is used in the paper for original training and fine-tuning.
3. Custom directories for data download and dataset storage for all datasets
4. MinMax transformer has also been added to `dc.molnet.chembl25_datasets`

**Featurizers**:
1. Safeguard against the NaN problem in SmilesToImage featurizer

**KerasModel**
1. Replaced the `print()` statements for losses with logger statements. A large fraction of DeepChem uses loggers, so this would make it more compatible

2. Losses were getting logged only while saving checkpoints. In general, one might want to keep them separate, since the frequencies of monitoring losses vs saving can be different then.

3. Added support for custom directories when saving checkpoints, or restoring existing model, or getting a list of checkpoints.

### CODE REVIEW

**Comment by VIGS25** [comment]
Travis Failure is unrelated.

**Comment by peastman** [comment]
> Replaced the print() statements for losses with logger statements.

I intentionally used print() here because the logger isn't working correctly, and we've never been able to figure out why.  See #1390.  If you can figure out how to fix that, we can switch over to logger.

**Comment by VIGS25** [comment]
I think there was a fix proposed for this in #1591, and I ran my jobs with logger and it works properly.

File: deepchem/models/keras_model.py [hunk 1]
```diff
@@ -922,12 +934,20 @@ def _exec_with_session(self, f):
...
+    model_dir: str, default None
+      Directory to get list of checkpoints from. Reverts to self.model_dir if None
+
+    """
+    if model_dir is None:
+      model_dir = self.model_dir
+    return tf.train.get_checkpoint_state(model_dir).all_model_checkpoint_paths
 
-  def restore(self, checkpoint=None):
+  def restore(self, checkpoint=None, restore_from=None):
```
Comment (peastman) [inline deepchem/models/keras_model.py]: The argument should be called `model_dir` for consistency with `save_checkpoint()` and `get_checkpoints()`.

**Comment by peastman** [comment]
Ok, if it's working now that's great.

File: deepchem/models/keras_model.py (hunk 1 above)
Comment (VIGS25) [reply deepchem/models/keras_model.py]: Done

File: deepchem/models/keras_model.py [hunk 2]
```diff
@@ -936,10 +956,14 @@ def restore(self, checkpoint=None):
       the path to the checkpoint file to load.  If this is None, the most recent
       checkpoint will be chosen automatically.  Call get_checkpoints() to get a
       list of all available checkpoints.
+    restore_from: str, default None
```
Comment (peastman) [inline deepchem/models/keras_model.py]: It still has the old argument name here.

**Comment by peastman** [comment]
One minor mistake in a docstring.  Otherwise it looks good.

File: deepchem/models/keras_model.py (hunk 2 above)
Comment (VIGS25) [reply deepchem/models/keras_model.py]: Fixed.

**Pull request merged**